### ISSUE TITLE
Fix weird kinetic platform behavior

### ISSUE THREAD

**Issue by MondayHopscotch** [bitDecayGames/Jump#40]
Sometimes platforms will get out of sync, this seems to be related to their speeds + possible rounding + ?

Ideally, if a platform is moving at the same speed as a separate platform, they should stay in sync indefinitely.

**Comment by Kenoshen** [comment]
Maybe use timing based on the node locations? I.e. 
'''
Init(){

totalTimeToFinishLapMillis = some millis
totalDistance = sum(dist(n, n+1), ...)
currentMillis = 0

}

getNewCurrentTime(delta){

currentMillis = (isInReverse ? currentMillis - delta : currentMillis + delta)

extra = 0
If (currentMillis > totalTimeToFinishLapMillis)
extra = currentMillis - totalTimeToFinishLapMillis
else if (currentMillis < 0)
extra = currentMillis

If (extra > 0 && !isBouncing)
currentMillis = extra
else if(extra > 0 && isBouncing)
isInReverse = true
currentMillis = totalTimeToFinishLapMillis - extra
else if (extra < 0 && !isBouncing)
currentMillis = totalTimeToFinishLapMillis + extra
else if (extra < 0 && isBouncing)
isInReverse = false
currentMillis = -(extra)

} 
'''
getPositionOnPath(){

currentPercentDone = currentMillis / totalTimeToFinishLapMillis

desiredDistance = totalDistance * currentPercentDone

findDist = 0
for (i = 1; i < nodes.len; i++){
findDist += dist(nodes[i - 1], nodes[i])
if findDist > desiredDistance
return (nodes[i]-nodes[i-1]).norm().multiply(findDist - desiredDistance)
}

}

**Comment by Kenoshen** [comment]
Sorry, those ''' were suppose to be backticks. I'm on my phone. Just ignore them.

**Comment by MondayHopscotch** [comment]
I spent some time talking about this with my cousin today shortly after putting this into here. It turns out it was just a stupid math error.

The way it currently is setting the velocity basically ends up with one dead from every time it touches a path node. This causes timings to go weird because each platform is losing 1 update worth of movement.

I tried a quick change to just add a vector based on the remaining distance it should be travelling and things seem correct. Will do some more testing before I claim things are fixed.
